src/main/scala/org/broadinstitute/dsde/workbench/gpalloc/dao/HttpGoogleBillingDAO.scala

```diff
@@ -462,10 +463,21 @@ class HttpGoogleBillingDAO(appName: String,
           case ge: GoogleJsonResponseException if ge.getStatusCode == 400 && ge.getMessage.contains("while it has other pending delete operations") =>
         }
       }
-    } yield {
-      //nah
-    }
-  }
+    } yield {}
+
+  // Leonardo currently only created GCE VMs in zone us-central1-a. If it were to start supporting other zones, this will need to be updated.
+  def cleanupVMs(projectName: String): Future[Unit] =
+    for {
+      result <- googleRq(computeManager.instances().list(projectName, "us-central1-a"))
+      instances = googNull(result.getItems)
+      instanceNames = instances.map(i => i.getName)
+      _ <- sequentially(instanceNames) { instanceName =>
```

Comment:
Also wondering if there is a reason we need to do this sequentially instead of `Future.traverse` 🤔

Reply:
if there are many `Future.traverse` may perform too many concurrent calls and hit quota issues. This is not particularly time sensitive so I wouldn't bother with paralellism.